### PR TITLE
tests, network: Remove SkipWhenNotDualStackCluster

### PR DESCRIPTION
**Pull request by oshoval**
### What this PR does
Use labels instead.
It statically determine which tests to run, without dynamic skipping.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/decorators/decorators.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ var (
 	USB                                  = []interface{}{Label("USB")}
 	AutoResourceLimitsGate               = []interface{}{Label("AutoResourceLimitsGate")}
 	RequiresTwoWorkerNodesWithCPUManager = []interface{}{Label("requires-two-worker-nodes-with-cpu-manager")}
+	DualStackCluster                     = []interface{}{Label("dual-stack-cluster")}
```
Comment (EdDev) [inline tests/decorators/decorators.go:53]: Aren't most of our lanes dual stack? I think the exception is to have a single stack case, no?
How about to place a label like `ipv6-single-stack` and `ipv4-single-stack`?

Comment (oshoval) [reply tests/decorators/decorators.go:53]: > Aren't most of our lanes dual stack? I think the exception is to have a single stack case, no?
correct 

> How about to place a label like ipv6-single-stack and ipv4-single-stack?

the test that is changed requires dual stack, so the minimal change that is needed is that if we are IPv6 single stack, skip dual stack tests
so we do need the label to be dual-stack, not x-single-stack
(PS note that we don't have a ipv4-single-stack kubevirtci variant)

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.29-ipv6-sig-network

lets see it runs the same amount of tests / pass

File: tests/decorators/decorators.go (hunk 1 above)
Comment (EdDev) [reply tests/decorators/decorators.go:53]: This is super confusing.

Then perhaps this name will fit better: `requires-dual-ip-stack-cluster`.

Comment (oshoval) [reply tests/decorators/decorators.go]: done with `requires-dual-stack-cluster ` hope its fine

**Comment by oshoval** [comment]
/hold 
need to fix something

**Comment by oshoval** [comment]
/hold cancel

Fix a mistake, the label should be on one of the Entries only (assuming we can add decorator to Entry as well, lets see the results)

Added commented as discussed

File: tests/network/expose.go [hunk 2]
```diff
@@ -321,7 +317,7 @@ var _ = SIGDescribe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:c
 			},
 				Entry("over SingleStack IP family policy", k8sv1.IPFamilyPolicySingleStack),
 				Entry("over PreferDualStack IP family policy", k8sv1.IPFamilyPolicyPreferDualStack),
-				Entry("over RequireDualStack IP family policy", k8sv1.IPFamilyPolicyRequireDualStack),
+				Entry("over RequireDualStack IP family policy", decorators.RequiresDualStackCluster, k8sv1.IPFamilyPolicyRequireDualStack),
```
Comment (xpivarc) [inline tests/network/expose.go:320]: What about the `Entry("over PreferDualStack IP family policy", k8sv1.IPFamilyPolicyPreferDualStack),`?

Comment (oshoval) [reply tests/network/expose.go:320]: it is checked already line above,
here we "enforce" dual

Comment (EdDev) [reply tests/network/expose.go:320]: > What about the `Entry("over PreferDualStack IP family policy", k8sv1.IPFamilyPolicyPreferDualStack),`?

If I’m not mistaken, that one does not “require” dualstack, it only “prefers” it.

**Review by EdDev — APPROVED** [review]
Nice, thanks!

**Comment by iholder101** [comment]
Thanks @oshoval!
/approve

File: tests/network/expose.go (hunk 2 above)
Comment (xpivarc) [reply tests/network/expose.go:320]: I am curios, how do you check the preference if you can't?

Comment (EdDev) [reply tests/network/expose.go:320]: Maybe I’m missing something, but in both cases it is checked, one when there is and the other when there is not.

The existing skip also treated the required scenario, not the preferred one.

Comment (oshoval) [reply tests/network/expose.go:320]: Now we would skip required only if the lanes aren't dual stack
all our vm based lanes beside one is dual stack
and we also have sanity check for that on kubevirtci check-up iirc

**Pull request merged**